projects/canopy/src/lib/forms/sort-code/sort-code.component.ts

```diff
@@ -0,0 +1,206 @@
+import {
+  Component,
+  ContentChild,
+  HostBinding,
+  Input,
+  OnInit,
+  Optional,
+  Self,
+  ViewChild,
+  ViewEncapsulation,
+} from '@angular/core';
+import {
+  ControlValueAccessor,
+  FormControl,
+  FormGroup,
+  FormGroupDirective,
+  NgControl,
+  ValidationErrors,
+  Validators,
+} from '@angular/forms';
+
+import { Subscription } from 'rxjs';
+import { filter } from 'rxjs/operators';
+
+import { LgErrorStateMatcher } from '../validation/error-state-matcher';
+import { LgDomService } from '../../utils/dom.service';
+import { LgHintComponent } from '../hint';
+import { SortCodeField } from './sort-code.interface';
+import { LgValidationComponent } from '../validation';
+
+let nextUniqueId = 0;
+
+@Component({
+  selector: 'lg-sort-code',
+  templateUrl: './sort-code.component.html',
+  styleUrls: ['./sort-code.component.scss'],
+  encapsulation: ViewEncapsulation.None,
+})
+export class LgSortCodeComponent implements OnInit, ControlValueAccessor {
+  @HostBinding('class.lg-sort-code-field') class = true;
+
+  private uniqueId = nextUniqueId++;
+  sortCodeFormGroup: FormGroup;
+  first: FormControl;
+  second: FormControl;
+  third: FormControl;
+  valueChanges: Subscription;
+  hasError: boolean;
+
+  @Input() value: string;
+  @Input() disabled: false;
+  @Input() firstId = `lg-input-sort-code-first-${this.uniqueId++}`;
```

Comment:
Not 100% sure about this. It's an approach copied from the Date Field component, so I assumed there was a reason that I hadn't thought of. I think I'd prefer to keep it consistent and raise an issue to remove if everyone thinks they're superfluous .

Reply:
they were left that way so that they could be overwritten if there was a need (copied from material ui). It seems unlikely but there could be a scenario where you want control over the id, for example interacting with a third party piece of code outside of Angular.